### ISSUE TITLE
state_transition should use SignedBeaconBlock

### ISSUE THREAD

**Issue by mratsim** [status-im/nim-beacon-chain#762]
Missed interface change in #647 

Normally an easy change (don't forget to update nbench #747 and the blockpool)

https://github.com/ethereum/eth2.0-specs/blob/v0.10.1/specs/phase0/beacon-chain.md#beacon-chain-state-transition-function

**Comment by JGcarv** [comment]
Hello @mratsim 

Is anyone working on this yet? Can I tackle it?

**Comment by arnetheduck** [comment]
All yours! We work off the [devel](https://github.com/status-im/nim-beacon-chain/blob/devel/beacon_chain/state_transition.nim#L223) branch for PR's. Let us know if you start working on it.

**Comment by JGcarv** [comment]
Good! I'll start later today. Thank you

**Comment by arnetheduck** [comment]
the PR highlights an interesting issue: the `invalid_block_sig` test is detecting a randao mismatch, so it doesn't detect that we're not validating the block signature - it passes though it shouldn't - this is a problem, because it meant that our test suite did not detect that we were not processing the block correctly - it needs further investigation.

https://github.com/ethereum/eth2.0-spec-tests/tree/master/tests/minimal/phase0/sanity/blocks/pyspec_tests/invalid_block_sig

**Comment by JGcarv** [comment]
I submitted a new PR finish the remaining work on this ticket.

Also, regarding the issue with `invalid_block_sig`, my guess is because randao is evaluated before the block sig on the sanity tests, and it was outdated causing it to always fail.

But it would be good to double check once `state_transition` is completed and #780 is merged!

**Comment by arnetheduck** [comment]
> Also, regarding the issue with invalid_block_sig, my guess is because randao is evaluated before the block sig on the sanity tests, and it was outdated causing it to always fail.

regarding this though, there are two possibilities: the randao is correct and if it wasn't for the signature, the test would pass - or the randao is broken, but nobody noticed because they have their sig check in order. I'd be curious to know which it is - if it's the latter with broken randao, it's not a good signature test really and we need to report it upstream.

**Comment by arnetheduck** [comment]
it should be possible to test this using https://github.com/protolambda/zcli

**Comment by tersec** [comment]
> it should be possible to test this using https://github.com/protolambda/zcli

If the last commit really is `v0.9.4 spec zrnt`, then not yet. https://github.com/protolambda/zrnt appears to be on 0.9.4 as well.

**Comment by arnetheduck** [comment]
obsolete
